docs/remote-k8s-access/README.md

```diff
@@ -0,0 +1,179 @@
+# Background
+K8ssandra Operator is capable of creating and managing a multi-datacenter Cassandra cluster that spans multiple Kubernetes clusters. The operator uses Kubernetes APIs to manage objects in remote clusters; therefore, it needs to create client connections with appropriate permissions to remote clusters.
+
+This document describes what is involved with creating and managing connections to remote clusters.
+
+When the operator starts up it queries for ClientConfig objects. A ClientConfig provides the necessary information for the operator to create client connections to remote clusters. 
+
+Before getting into details of using ClientConfigs we need to provide some background on service accounts, kubeconfig files, and secrets.
+
+## ServiceAccount
+All pods are associated with a ServiceAccount. If no ServiceAccount is specified for a pod, then the default one will be used.
+
+**Note:** Kubernetes creates a default ServiceAccount in each namespace.
+
+The ServiceAccount is used to authenticate and authorizate the pod for any operations that it wants to perform against the api server. Authentication is done by way of a service account token which is stored in a secret. Kubernetes automatically mounts the secrect in the pod. Authorization is done with Roles and RoleBindings. The ServiceAccount is mapped to Roles and RoleBindings with RoleBindings and ClusterRoleBindings respectively.
```

Comment:
```suggestion
The ServiceAccount is used to authenticate and authorize the pod for any operations that it wants to perform against the api server. Authentication is done by way of a service account token which is stored in a secret. Kubernetes automatically mounts the secret in the pod. Authorization is done with Roles and RoleBindings. The ServiceAccount is mapped to Roles and RoleBindings with RoleBindings and ClusterRoleBindings respectively.
```

Reply:
This sentence is a bit confusing imo:

> The ServiceAccount is mapped to Roles and RoleBindings with RoleBindings and ClusterRoleBindings respectively.

I am not sure to understand what is mapped to what.

Reply:
I removed it and shortened the section. There are three things I want to point out in the section - 1) a service account is used for auth for pods, 2) a service account is authenticated with a token, and 3) the token is stored in a secret.